invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -77,6 +77,19 @@ def get_type(self, obj):
         props = resource_type_record["props"]
         return props.get("csl", "article")  # article is CSL "Other"
 
+    def get_doi(self, obj):
+        """Get doi."""
+        doi = obj["pids"].get("doi")
+
+        if doi:
+            doi = doi["identifier"]
+        else:
+            identifiers = obj["metadata"].get("identifiers")
+            for identifier in identifiers:
+                if identifier["scheme"] == "doi":
+                    doi = identifier["identifier"]
+        return doi
```

Comment:
```suggestion
        if doi:
            return doi["identifier"]
        
        identifiers = obj["metadata"].get("identifiers", [])
        for identifier in identifiers:
            if identifier["scheme"] == "doi":
                return identifier["identifier"]
        
        return ""
```
I am not 100% on the last line -- what used to happen when there were no `pids.doi.identifier` before for the citation? What should happen when there are no `pids.doi.identifier` _and_ no alternative? Should we return empty string or None?

Reply:
Thanks for the improvement-that's much clearer. I used 'missing' for the last line, as that's used elsewhere in the serializer and behaves as expected (the doi field doesn't show up in the CSL if it's not there)